packages/phoenix-evals/src/phoenix/evals/classify.py

```diff
@@ -164,43 +200,56 @@ def _process_response(response: str) -> Tuple[str, Optional[str]]:
             unrailed_label, explanation = parse_openai_function_call(response)
         return snap_to_rail(unrailed_label, rails, verbose=verbose), explanation
 
-    async def _run_llm_classification_async(prompt: str) -> ParsedLLMResponse:
+    async def _run_llm_classification_async(input_data: pd.Series) -> ParsedLLMResponse:
         with set_verbosity(model, verbose) as verbose_model:
+            prompt = _map_template(input_data)
             response = await verbose_model._async_generate(
                 prompt, instruction=system_instruction, **model_kwargs
             )
         inference, explanation = _process_response(response)
-        return inference, explanation, response
+        return inference, explanation, response, prompt
 
-    def _run_llm_classification_sync(prompt: str) -> ParsedLLMResponse:
+    def _run_llm_classification_sync(input_data: pd.Series) -> ParsedLLMResponse:
         with set_verbosity(model, verbose) as verbose_model:
+            prompt = _map_template(input_data)
             response = verbose_model._generate(
                 prompt, instruction=system_instruction, **model_kwargs
             )
         inference, explanation = _process_response(response)
-        return inference, explanation, response
+        return inference, explanation, response, prompt
 
-    fallback_return_value: ParsedLLMResponse = (None, None, "")
+    fallback_return_value: ParsedLLMResponse = (None, None, "", "")
 
     executor = get_executor_on_sync_context(
         _run_llm_classification_sync,
         _run_llm_classification_async,
         run_sync=run_sync,
         concurrency=concurrency,
         tqdm_bar_format=tqdm_bar_format,
-        exit_on_error=True,
+        max_retries=max_retries,
+        exit_on_error=exit_on_error,
         fallback_return_value=fallback_return_value,
     )
 
-    results = executor.run(prompts.tolist())
-    labels, explanations, responses = zip(*results)
+    results, execution_details = executor.run([row_tuple[1] for row_tuple in dataframe.iterrows()])
+    labels, explanations, responses, prompts = zip(*results)
+    all_exceptions = [s.exceptions for s in execution_details]
+    execution_statuses = [s.status for s in execution_details]
```

Comment:
Suggestion: Assuming `s` stands for "status" here, but maybe `details` is more appropriate given the type we are iterating over.